### ISSUE TITLE
Extending support to `biopython>1.79`

### ISSUE THREAD

**Issue by ojeda-e** [prody/ProDy#1723]
Hello ProDy developers. 

I have been working with your package and we are trying to stay updated with dependencies. Recently I bumped ProDy to 2.4.0 and found out that it depends on `biopython<=1.79`. Effectively, in your `setup.py` biopython is restricted to `<=1.79` (added in https://github.com/prody/ProDy/commit/d65a4c1dc6854fd35c568eafc57bb62718607506).

I am certain you have strong reasons to pin the version of Biopython. However, I was wondering if ProDy devs have any plans of extending support for ProDy in higher versions of Biopython in the short term. Thanks!

**Comment by jamesmkrieger** [comment]
Hello, yes, I think this should be fine. I think the main thing that affected us in the shift to 1.80 was the loss of pairwise2 but we now have a way to handle that, so it should be fine. We’ll have to try it and see.

**Comment by jamesmkrieger** [comment]
Of course, everything is always more complicated than one would think. The output from alignment has some different format in later biopython versions so we'll have to look into what the differences are and change some functions that use it.

**Comment by merkys** [comment]
I tried building bb2b63f7156eeb3cfbfc2e8eba4e68bf26036bb0 with biopython 1.81 and it failed `testBuildMSAlocal` with the following on both Python 3.11 and 3.12:
```
======================================================================
FAIL: testBuildMSAlocal (prody.tests.sequence.test_analysis.TestBuildMSA.testBuildMSAlocal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/merkys/ProDy/.pybuild/cpython3_3.11_prody/build/prody/tests/sequence/test_analysis.py", line 1210, in testBuildMSAlocal
    assert_array_equal(expect, result)
  File "/usr/lib/python3/dist-packages/numpy/testing/_private/utils.py", line 985, in assert_array_equal
    assert_array_compare(operator.__eq__, x, y, err_msg=err_msg,
  File "/usr/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/numpy/testing/_private/utils.py", line 862, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

Mismatched elements: 2 / 2 (100%)
 x: array([<Sequence: 1 (msa_3hsyA_3o21A[0]; length 380; 354 residues and 26 gaps)>,
       <Sequence: 2 (msa_3hsyA_3o21A[1]; length 380; 374 residues and 6 gaps)>],
      dtype=object)
 y: array([<Sequence: 1 (Unknown[0]; length 80; 59 residues and 21 gaps)>,
       <Sequence: 2 (Unknown[1]; length 80; 64 residues and 16 gaps)>],
      dtype=object)
```

**Comment by merkys** [comment]
Sorry, I assumed #1816 is merged, will try building #1816.

**Comment by jamesmkrieger** [comment]
> Sorry, I assumed #1816 is merged, will try building #1816.

ok, sounds good.

I'm sorry that it isn't yet. I was waiting for someone to try writeMMTF

**Comment by jamesmkrieger** [comment]
I've tested writeMMTF enough now and merged it
